integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateTableHelper.scala

```diff
@@ -166,8 +166,9 @@ case class PreAggregateTableHelper(
     // table.
     SegmentStatusManager.deleteLoadsAndUpdateMetadata(parentTable, false)
     val loadAvailable = SegmentStatusManager.readLoadMetadata(parentTable.getMetadataPath)
-    if (loadAvailable.exists(load => load.getSegmentStatus == SegmentStatus.INSERT_IN_PROGRESS ||
-      load.getSegmentStatus == SegmentStatus.INSERT_OVERWRITE_IN_PROGRESS)) {
+    if (loadAvailable.exists(load => load.getVisibility().equalsIgnoreCase("true") &&
```

Comment:
Please use the utility SegmentStatusManager.isLoadInProgressInTable

Reply:
Done